File: 0129-sum-root-to-leaf-numbers/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int cal(TreeNode root,int sum){
        if(root == null)return 0;
        
        sum = sum*10 + root.val;
        if(root.left==null && root.right==null)
        {
            return sum;
        }
        return cal(root.left,sum)+cal(root.right,sum);
    }
    public int sumNumbers(TreeNode root) {
        return cal(root,0);
    }
}

File: 0129-sum-root-to-leaf-numbers/solution.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
void sum(struct TreeNode* root,int cr,int* psum){
    if(root==NULL)return ;
    if(root->left ==NULL && root->right==NULL){
        *psum += (cr*10) +root->val;
        return;
    }
    sum(root->left, cr*10+root->val , psum);
    sum(root->right, cr*10+root->val , psum);
}
int sumNumbers(struct TreeNode* root) {
   int psum=0;
   sum(root,0,&psum);
   return psum; 
}
